tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+    found in the list of active support cases, notify the user.
+
+    Parameters
+    ----------
+    channel_id : str
+        unique string used to idenify a Slack channel. Used to send messages to the channel
+    channel_name : str
+        designated channel name of the channel. For users to understand where their
+        cases are being tracked in Slack
+    case : str
+        unique id of the case
+    user_id : str
+        the Slack user_id of the user who submitted the request. Used to send ephemeral
+        messages to the user
+    """
+    parent = get_parent(case)
+    if parent == 'Case not found':
+        case_not_found(channel_id, user_id, case)
+    else:
+        tracker = {
+            "channel_id": channel_id, 
+            "case": case,
+            "channel_name": channel_name
+        }
+
+        exists = False
+    
+        for tc in tracked_cases:
+            if tc['channel_id'] == channel_id and tc['case'] == case:
+                exists = True
+                break
+
+        if exists == False:
+            tracked_cases.append(tracker)
+            
+            try:
+                file_overwrite(tracked_cases_file, tracked_cases)
+            except OSError as e:
+                logging.error(e, ' : {}'.format(datetime.now()))
+                time.sleep(0.1)
+                file_overwrite(tracked_cases_file, tracked_cases)
+
+            client.chat_postMessage(channel=channel_id, text=f"{channel_name} is now tracking case {case}")
+        else:
+            client.chat_postEphemeral(channel=channel_id, user=user_id, text=f"Case {case} is already being tracked in {channel_name}")
+
+
+def add_comment(channel_id, case, comment, user_id, user_name):
+    """
+    Add a comment to a Google Cloud support case.
+    
+    Parameters
+    ----------
+    channel_id : str
+        unique string used to idenify a Slack channel. Used to send messages to the channel
+    case : str
+        unique id of the case
+    comment : str
+        comment to be added to the case
+    user_id : str
+        the Slack user_id of the user who submitted the request. Used to send ephemeral
+        messages to the user
+    user_name : str
+        Slack user_name of the user that ran the command. Appended to the end of the
+        comment to identify who submitted submitted it, otherwise all comments will
+        show as coming from the case creator
+    """
+    client.chat_postEphemeral(channel=channel_id, user=user_id, text="Your request is processing ...")
+    parent = get_parent(case)
+    
+    if parent == 'Case not found':
+        case_not_found(channel_id, user_id, case)
+    else:
+        req_body = {
+            "body" : comment + '\n*Comment submitted by {} via Google Cloud Support Slack bot*'.format(user_name)
+        }
+        req = support_service.cases().comments().create(parent=parent, body=req_body)
+        try:
+            req.execute()          
```

Comment:
Can you add max_retires to improve reliability?

Reply:
Added